src/main/java/fi/metatavu/kuntaapi/server/integrations/gtfs/GtfsPublicTransportProvider.java

```diff
@@ -0,0 +1,209 @@
+      return null;
+    }
+    
+    return gtfsPublicTransportRouteCache.get(routeId);
+  }
+  
+  @Override
+  public List<Stop> listStops(OrganizationId organizationId) {
+    List<PublicTransportStopId> stopIds = identifierRelationController.listPublicTransportStopIdsBySourceAndParentId(GtfsConsts.IDENTIFIER_NAME, organizationId);
+    List<Stop> stops = new ArrayList<>(stopIds.size());
+    
+    for (PublicTransportStopId stopId : stopIds) {
+      Stop stop = gtfsPublicTransportStopCache.get(stopId);
+      if (stop != null) {
+        stops.add(stop);
+      }
+    }
+    
+    return stops;
+  }
+
+  @Override
+  public Stop findStop(OrganizationId organizationId, PublicTransportStopId stopId) {
+    if (!identifierRelationController.isChildOf(organizationId, stopId)) {
+      return null;
+    }
+    
+    return gtfsPublicTransportStopCache.get(stopId);
+  }
+  
+  @Override
+  public List<StopTime> listStopTimes(OrganizationId organizationId, PublicTransportStopId stopId, Integer departureTime) {
+    List<PublicTransportStopTimeId> stopTimeIds = identifierRelationController.listPublicTransportStopTimeIdsBySourceAndParentId(GtfsConsts.IDENTIFIER_NAME, organizationId);
+    List<StopTime> stopTimes = new ArrayList<>(stopTimeIds.size());
+    
+    for (PublicTransportStopTimeId stopTimeId : stopTimeIds) {
+      StopTime stopTime = gtfsPublicTransportStopTimeCache.get(stopTimeId);
+      if (stopTime != null && isAcceptableStopTime(stopTime, stopId, departureTime)) {
+        stopTimes.add(stopTime);
+      }
+    }
+    
+    return stopTimes;
+  }
+
+  @Override
+  public StopTime findStopTime(OrganizationId organizationId, PublicTransportStopTimeId stopTimeId) {
+    if (!identifierRelationController.isChildOf(organizationId, stopTimeId)) {
+      return null;
+    }
+    
+    return gtfsPublicTransportStopTimeCache.get(stopTimeId);
+  }
+  
+  @Override
+  public List<Trip> listTrips(OrganizationId organizationId) {
+    List<PublicTransportTripId> tripIds = identifierRelationController.listPublicTransportTripIdsBySourceAndParentId(GtfsConsts.IDENTIFIER_NAME, organizationId);
+    List<Trip> trips = new ArrayList<>(tripIds.size());
+    
+    for (PublicTransportTripId tripId : tripIds) {
+      Trip trip = gtfsPublicTransportTripCache.get(tripId);
+      if (trip != null) {
+        trips.add(trip);
+      }
+    }
+    
+    return trips;
+  }
+
+  @Override
+  public Trip findTrip(OrganizationId organizationId, PublicTransportTripId tripId) {
+    if (!identifierRelationController.isChildOf(organizationId, tripId)) {
+      return null;
+    }
+    
+    return gtfsPublicTransportTripCache.get(tripId);
+  }
+
+  private boolean isAcceptableStopTime(StopTime stopTime, PublicTransportStopId stopId, Integer departureTime) {
+    if (stopId != null && !stopId.getId().equals(stopTime.getStopId())) {
+      return false;
+    }
+    
+    if (departureTime != null && stopTime.getDepartureTime() < departureTime) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)